src/sema/ast.rs

```diff
@@ -56,11 +56,30 @@ pub enum Type {
     Void,
     Unreachable,
     /// DynamicBytes and String are lowered to a vector.
-    Slice,
+    Slice(Box<Type>),
     /// We could not resolve this type
     Unresolved,
 }
 
+#[derive(PartialEq, Clone, Eq, Hash, Debug)]
+pub enum ArrayLength {
+    Fixed(BigInt),
+    Dynamic,
+    /// Any fixed-length allowed here. This is used as parameter type for some
+    /// system calls, sol_create_program_address() for example.
```

Comment:
I think it is useful to give an example of a AnyFixed length. I still couldn't grasp the difference between `AnyFixed` and `Fixed`, from your explanation.

Reply:
I've re-written the comment. Hope you find this clearer